contrib/syntax/vim/syntax/dockerfile.vim

```diff
@@ -27,5 +27,5 @@ let s:current_syntax = b:current_syntax
 unlet b:current_syntax
 syntax include @SH syntax/sh.vim
 let b:current_syntax = s:current_syntax
-syntax region shLine matchgroup=dockerfileKeyword start=/\v^\s*(RUN|CMD|ENTRYPOINT)\s/ end=/\v$/ contains=@SH
+syntax region shLine matchgroup=dockerfileKeyword start=/\v^\s*%(ONBUILD\s+)?%(RUN|CMD|ENTRYPOINT)\s/ end=/\v$/ contains=@SH
```

Comment:
This change LGTM (assuming we can clarify what the `%` around it are for / point to the docs for where that's described here in the PR discussion).